enhancements/single-node/single-node-openshift-day2-workers.md

```diff
@@ -0,0 +1,453 @@
+the user with worker node ignition files. As a result, the user is not able to
+manually add worker nodes to the resulting cluster. All current official OCP
+documentation points the user to use "the worker ignition files that get
+generated during installation", but these are not currently generated in such
+bootstrap-in-place installation.
+
+### Goals
+
+- Define a sensible value for the "Infrastructure Topology" parameter in the
+various configurations a single control-plane node cluster may be deployed in.
+
+- Deal with the "floating ingress" issue encountered in "none"-platform
+single control-plane node clusters which have worker nodes added to them.
+
+- Define how the installer may be modified to generate the worker ignition
+manifest, even when doing bootstrap-in-place "none"-platform single
+control-plane node cluster installation
+
+### Non-Goals
+
+- Deal with clusters that have less nodes than they had during day-1
+installation. The rationale for this being a non-goal is that it seems that
+removing the "initial" nodes of a cluster causes a lot of issues even in a
+regular, multi-node installation. So dealing with similar issues in single
+control-plane node clusters is also out of scope for this enhancement. For
+example, installing a cluster with 3 control-plane nodes and 3 workers results
+in unschedulable control-plane nodes. Scaling down the worker machine-sets from
+3 to 0 post-installation does not magically make the control-plane nodes
+schedulable, and as a result a lot of infrastructure workloads fail to schedule
+and operators degrade. User intervention is required to fix this.
+
+- Deal with expansion of the single-node control-plane by adding more control-plane nodes
+
+- Deal with expansion of clusters that have been installed before the
+implementation of this enhancement (if possible, their expansion may addressed
+of with documentation outlining the steps that the user has to take to enable
+expansion).
+
+- Adjust the baremetal platform to support single control-plane node
+installations. The baremetal platforms solves the "floating ingress" issue by
+using virtual IP addresses/keepalived.
+
+- Deal with users who want their ingress traffic to not go through the single
+control-plane node.
+
+- Deal with non-cloud, non-"none" platforms such as baremetal, vSphere, etc.
+
+## Proposal
+
+Set the "Infrastructure Topology" value to "SingleReplica" in almost all 
+circumstances where the "Control Plane Topology" is also "SingleReplica".
+
+Make sure the default `IngressController` CR created by the Cluster Ingress
+Operator targets the "master" pool rather than the "worker" pool whenever the
+"Infrastructure Topology" is set to "SingleReplica". This will ensure that the
+`router-default` deployment created by the Cluster Ingress Operator will always
+run on the single control plane node, and as a result any
+`*.apps.<cluster>.<base>` DNS entries which originally pointed at the single
```

Comment:
How worker will have those dns values? or we don't need them for workers?

Reply:
We do need them for workers. The user is supposed to make sure DNS is configured "globally" and pointing to the control-plane node before trying to add workers. 
In the case of Assisted, we'll do our best to do it automatically whenever possible with dnsmasq tricks.